tracer/build/_build/Build.GitHub.cs

```diff
@@ -496,6 +496,7 @@ partial class Build
                                            buildId: build.Id,
                                            artifactName: artifactName);
 
+                            commitShaWithArtifacts = commitSha;
```

Comment:
nit: I don't think you need to introduce a new variable here - define `commitSha` outside the loop (where you're currently defining `commitShaWithArtifacts`).  If we successfully exit this loop, `commitSha` will already be set correctly.

Reply:
Done in 1ad8ac42a336fecd7aca60a84fe6b98aca0109d7